ddtrace/tracer/tracer.go

```diff
@@ -299,10 +303,13 @@ func (t *tracer) flushTraces() {
 	if t.config.debug {
 		log.Printf("Sending payload: size: %d traces: %d\n", size, count)
 	}
-	err := t.config.transport.send(t.payload)
+	rc, err := t.config.transport.send(t.payload)
 	if err != nil {
 		t.pushError(&dataLossError{context: err, count: count})
 	}
+	if err == nil && t.config.prioritySampling != nil {
+		t.config.prioritySampling.readRatesJSON(rc) // TODO: handle error?
```

Comment:
I think we should not handle this kind of error (it will be a mismatch in what the Agent returns) and so we should just bail out and log. Fine to leave the `TODO` because we're going to rethink the Tracer logging to avoid spam in users' logs.

Reply:
I was thinking along the same lines, good to have the todo there though